src/hb-ot-name-table.hh

```diff
@@ -97,12 +97,34 @@ struct NameRecord
     return UNSUPPORTED;
   }
 
-  NameRecord* copy (hb_serialize_context_t *c, const void *base) const
+  NameRecord* copy (hb_serialize_context_t *c, const void *base,
+                    const hb_hashmap_t<unsigned, const char*> *name_table_overrides) const
   {
     TRACE_SERIALIZE (this);
+    auto snap = c->snapshot ();
     auto *out = c->embed (this);
     if (unlikely (!out)) return_trace (nullptr);
-    out->offset.serialize_copy (c, offset, base, 0, hb_serialize_context_t::Tail, length);
+    if (name_table_overrides->has (nameID)) {
+      const char *name_str = name_table_overrides->get (nameID);
+      unsigned text_size = 4 * strlen (name_str);
+      char *name_str_utf16_be = (char *) hb_calloc (text_size + 1, HBUINT16::static_size);
```

Comment:
text_size is the number of bytes, so divide it by two since we're allocating a number of HBUINT16's here.

Reply:
Done, I made it = text_size/2 +1 to save room for NULL terminator.